src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -484,6 +484,40 @@ async Task<DurableOrchestrationStatus> IDurableOrchestrationClient.GetStatusAsyn
             return await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput, showInput);
         }
 
+        /// <inheritdoc />
+        async Task<List<DurableOrchestrationStatus>> IDurableOrchestrationClient.GetStatusAsync(IEnumerable<string> instanceID, bool showHistory, bool showHistoryOutput, bool showInput)
+        {
+            var results = new List<DurableOrchestrationStatus>();
+            foreach (string id in instanceID)
+            {
+                IList<OrchestrationState> stateList;
+                try
+                {
+                    stateList = await this.DurabilityProvider.GetOrchestrationStateWithInputsAsync(id, showInput);
+                }
+                catch (NotImplementedException)
+                {
+                    // TODO: Ignore the show input flag. Should consider logging a warning.
+                    stateList = await this.client.ServiceClient.GetOrchestrationStateAsync(id, allExecutions: false);
+                }
+
+                OrchestrationState state = stateList?.FirstOrDefault();
+                var currenStatus = new DurableOrchestrationStatus();
+                if (state == null || state.OrchestrationInstance == null)
+                {
+                    currenStatus = null;
+                }
+                else
+                {
+                    currenStatus = await GetDurableOrchestrationStatusAsync(state, this.client, showHistory, showHistoryOutput, showInput);
+                }
+
+                results.Add(currenStatus);
+            }
```

Comment:
The code here seems to be a duplicate of the code within `GetStatusAsync(string instanceId, bool showHistory, bool showHistoryOutput, bool showInput)`. Could we replace most of this with just a call to that other method?

Put differently, I think this code would be a bit simpler if we just had a for-loop that calls onto the `GetStatusAsync` that takes 1 instanceId at a time. Is there any reason not to do that?